Tests/xSharePoint/xSharePoint.xSPSearchTopology.Tests.ps1

```diff
@@ -0,0 +1,307 @@
+                Add-Type -TypeDefinition "public class QueryProcessingComponent { public string ServerName { get; set; } public System.Guid ComponentId {get; set;}}"
+                Add-Type -TypeDefinition "public class IndexComponent { public string ServerName { get; set; } public System.Guid ComponentId {get; set;} public System.Int32 IndexPartitionOrdinal {get; set;}}"
+
+                $adminComponent = New-Object AdminComponent
+                $adminComponent.ServerName = $env:COMPUTERNAME
+                $adminComponent.ComponentId = [Guid]::NewGuid()
+
+                $crawlComponent = New-Object CrawlComponent
+                $crawlComponent.ServerName = $env:COMPUTERNAME
+                $crawlComponent.ComponentId = [Guid]::NewGuid()
+
+                $contentProcessingComponent = New-Object ContentProcessingComponent
+                $contentProcessingComponent.ServerName = $env:COMPUTERNAME
+                $contentProcessingComponent.ComponentId = [Guid]::NewGuid()
+
+                $analyticsProcessingComponent = New-Object AnalyticsProcessingComponent
+                $analyticsProcessingComponent.ServerName = $env:COMPUTERNAME
+                $analyticsProcessingComponent.ComponentId = [Guid]::NewGuid()
+
+                $queryProcessingComponent = New-Object QueryProcessingComponent
+                $queryProcessingComponent.ServerName = $env:COMPUTERNAME
+                $queryProcessingComponent.ComponentId = [Guid]::NewGuid()
+
+                $indexComponent = New-Object IndexComponent
+                $indexComponent.ServerName = $env:COMPUTERNAME
+                $indexComponent.IndexPartitionOrdinal = 0
+
+                return @($adminComponent, $crawlComponent, $contentProcessingComponent, $analyticsProcessingComponent, $queryProcessingComponent, $indexComponent)
+            }
+
+            Mock Get-SPEnterpriseSearchServiceInstance  {
+                return @{
+                    Server = @{
+                        Address = $env:COMPUTERNAME
+                    }
+                    Status = "Online"
+                }
+            }
+
+            $testParams = @{
+                ServiceAppName          = "Search Service Application"
+                Admin                   = @($env:COMPUTERNAME)
+                Crawler                 = @($env:COMPUTERNAME)
+                ContentProcessing       = @($env:COMPUTERNAME)
+                AnalyticsProcessing     = @($env:COMPUTERNAME)
+                QueryProcessing         = @($env:COMPUTERNAME)
+                IndexPartition          = @($env:COMPUTERNAME)
+                FirstPartitionDirectory = "I:\SearchIndexes\0"
+            }
+
+            It "returns true from the test method" {
+                Test-TargetResource @testParams | Should Be $true
+            }
+        }
+    }
+}
```

Comment:
OK well if there is no topology it will add one, and i've tested it to update topologies that don't match in a number of component types and it appears to play nice making those changes. So from what I've seen in here it should be fine at enforcing an existing topology. But you are right, it won't fail gracefully if there is no search service app that it finds so I'll code that up and push it in here. Will let you know when it's done mate.